### PR TITLE
[Golang] Assignment-1 : Implement linux shell

### PR DESCRIPTION
**Pull request by priyanka25081999**
Problem Statement:
Implement the simple linux interactive shell that supports commands like ls, pwd, exit, cd, history.

Solution:
https://github.com/velotio-tech/go-k8s-training/pull/65#issuecomment-1296908496

Signed-off-by: Priyanka Salunke <priyanka.salunke@velotio.com>

### CODE REVIEW

**Comment by priyanka25081999** [comment]
Execution:

```
priyankasalunke@velotio-ThinkPad-E14:~/velotio/go-k8s-training/assignment-solutions$ go run main.go commands.go 
*** Linux Interative Shell ***

priyankasalunke@velotio-ThinkPad-E14 ~/velotio/go-k8s-training/assignment-solutions$ pwd

Executing command :pwd
-----------------------
pwd : Current working directory :  /velotio/go-k8s-training/assignment-solutions

priyankasalunke@velotio-ThinkPad-E14 ~/velotio/go-k8s-training/assignment-solutions$ ls /home

Executing command :ls
-----------------------
Path : /home
.ecryptfs
priyankasalunke
priyankasalunke.unKDLgHn
velotio

priyankasalunke@velotio-ThinkPad-E14 ~/velotio/go-k8s-training/assignment-solutions$ history

Executing command :history
-----------------------
1 pwd
2 ls
3 history

priyankasalunke@velotio-ThinkPad-E14 ~/velotio/go-k8s-training/assignment-solutions$ cd /velotio/go-k8s-training

Executing command :cd
-----------------------
priyankasalunke@velotio-ThinkPad-E14 ~/velotio/go-k8s-training$ pwd

Executing command :pwd
-----------------------
pwd : Current working directory :  /velotio/go-k8s-training

priyankasalunke@velotio-ThinkPad-E14 ~/velotio/go-k8s-training/assignment-solutions$ exit

Executing command :exit
-----------------------

Exiting from the shell. Thank you!
```

File: assignment-solutions/commands.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	return dir
+}
+
+// Change the working directory
+func processChangeWorkingDir(args []string) {
+	if len(args) == 0 {
+		fmt.Println("Error : Invalid command argument!")
+	}
```
Comment (prafull11) [inline assignment-solutions/commands.go:84]: If no args you should go to home directory.

Comment (prafull11) [reply assignment-solutions/commands.go:84]: If more than one args then return error

File: assignment-solutions/commands.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+		if err != nil {
+			fmt.Println("Error : no path found!")
+			return
+		}
+
+		file, _ := os.ReadDir(path)
+		for _, file := range file {
+			fmt.Println(file.Name())
+		}
```
Comment (prafull11) [inline assignment-solutions/commands.go:121]: You should have read it in files not in file object and the print should be common for both conditions `ls` and `ls ~/path`

**Comment by prafull11** [comment]
Some comments to just look at. Rest looks good

File: assignment-solutions/commands.go (hunk 1 above)
Comment (priyanka25081999) [reply assignment-solutions/commands.go:84]: Sure

**Pull request closed without merging**